src/components/Charts/ChartFutureHospitalization.tsx

```diff
@@ -0,0 +1,239 @@
+import { AxisBottom, AxisLeft } from '@vx/axis';
+import { LinePath } from '@vx/shape';
+import { ParentSize } from '@vx/responsive';
+import { scaleLinear, scaleTime } from '@vx/scale';
+import { Projections } from 'common/models/Projections';
+import { COLORS } from 'common';
+import BoxedAnnotation from './BoxedAnnotation';
+import ChartContainer from './ChartContainer';
+import RectClipGroup from './RectClipGroup';
+import Tooltip from './Tooltip';
+import { LegendMarker, LegendLine } from './Legend';
+import * as Style from './Charts.style';
+import { formatDate, formatInteger } from './utils';
+import palette from 'assets/theme/palette';
+
+type Point = {
+  x: number;
+  y: number;
+};
+
+type PointTooltip = Point & {
+  color: string;
+  name: string;
+};
+
+const getDate = (p: Point) => new Date(p.x);
+const getY = (p: Point) => p.y;
+const hasData = (d: any) => isDate(getDate(d)) && Number.isFinite(getY(d));
+const getLastPoint = (points: Point[]): Point => last(points) || { x: 0, y: 0 };
+const isFuture = (p: Point) => getDate(p) > new Date();
+
+const getTooltipPoint = (
+  points: Point[],
+  color: string,
+  name: string,
+): PointTooltip[] => points.map(p => ({ ...p, color, name }));
+
+const getTooltipBody = (p: PointTooltip): ReactNode => (
+  <span>
+    <b>{formatInteger(getY(p))}</b>{' '}
+    {p.name === 'beds' ? 'beds available on' : 'hospitalizations expected by'}{' '}
+    <b>{formatDate(getDate(p), 'MMMM D')}</b>
+  </span>
+);
+
+const ChartFutureHospitalization = ({
+  projections,
+  width,
+  height,
+  marginTop = 5,
+  marginBottom = 40,
+  marginLeft = 48,
+  marginRight = 5,
+}: {
+  projections: Projections;
+  width: number;
+  height: number;
+  marginTop?: number;
+  marginBottom?: number;
+  marginLeft?: number;
+  marginRight?: number;
+}) => {
+  const chartWidth = width - marginLeft - marginRight;
+  const chartHeight = height - marginTop - marginBottom;
+
+  const dataNoAction = projections.baseline.getDataset('hospitalizations');
+  const dataNoActionFuture: Point[] = dataNoAction
+    .filter(hasData)
+    .filter(isFuture);
+
+  const dataProjected = projections.projected.getDataset('hospitalizations');
+  const dataProjectedFuture: Point[] = dataProjected
+    .filter(hasData)
+    .filter(isFuture);
+  const dataProjectedPast: Point[] = dataProjected
+    .filter(hasData)
+    .filter((p: Point) => !isFuture(p));
+
+  const dataBeds = projections.primary.getDataset('beds');
+
+  const allData: PointTooltip[] = [
+    ...getTooltipPoint(dataProjectedFuture, COLORS.PROJECTED, 'projected'),
+    ...getTooltipPoint(dataNoActionFuture, COLORS.LIMITED_ACTION, 'no-action'),
+    ...getTooltipPoint(dataProjectedPast, palette.black, 'past'),
+    ...getTooltipPoint(dataBeds, palette.chart.axis, 'beds'),
+  ];
+
+  const dateTwoWeeks = moment().add(2, 'weeks').toDate();
+  const minDate = d3min(allData, getDate) || new Date('2020-01-01');
+  const maxDate = d3max(allData, getDate) || dateTwoWeeks;
```

Comment:
To be clear, these min/max dates should never get used, right?  If there's no data, then the chart will be empty and other stuff will probably break (e.g. firstPointBeds and lastPastPoint).

So I'd be a little tempted to do:
```
  const minDate = d3min(allData, getDate);
  const maxDate = d3max(allData, getDate);
  assert(minDate && maxDate, 'Data must not be empty');
```

(make sure to import assert from 'common/utils')

Reply:
I added the defaults to TS won't complain - if I don't, it says that `minDate` and `maxDate` could be `undefined`. Would the `assert` get rid of the warnings? I'm not sure what the best practices are

Reply:
Yeah, if you use the assert() from `common/utils` it'll get rid of the warning:

```
export function assert(condition: any, msg?: string): asserts condition {
  if (!condition) {
    fail(msg);
  }
}
```

The `asserts condition` part basically tells TypeScript that the condition will be true if the function returns and so TypeScript will stop complaining.